storagenode/piecestore/endpoint.go

```diff
@@ -702,5 +703,9 @@ func ignoreEOF(err error) error {
 	if errs.Is(err, io.EOF) {
 		return nil
 	}
+	// Ignore if the client has already closed the connection.
+	if errs.Is(err, syscall.EPIPE) {
```

Comment:
does this only happen on a subset of situations where ignoreEOF is called? i think i'd prefer we have ignoreEOF and ignorePIPE as separate functions, so we can add ignorePIPE in only the cases we know we need it.